declaration.dd

```diff
@@ -325,8 +328,12 @@ $(GNAME AliasDeclaration):
     $(D alias) $(I AliasDeclarationX) $(D ;)
 
 $(GNAME AliasDeclarationX):
+    $(GLINK AliasDeclarationY)
+    $(I AliasDeclarationX) $(D ,) $(GLINK AliasDeclarationY)
+
+$(GNAME AliasDeclarationY):
     $(I Identifier) $(GLINK2 template, TemplateParameters)$(OPT) $(D =) $(GLINK StorageClasses)$(OPT) $(GLINK Type)
-    $(I AliasDeclarationX) $(D ,) $(I Identifier) $(GLINK2 template, TemplateParameters)$(OPT) $(D =) $(GLINK StorageClasses)$(OPT) $(GLINK Type)
+    $(I Identifier) $(GLINK2 template, TemplateParameters)$(OPT) $(D =) $(GLINK2 expression, FunctionLiteral)
```

Comment:
To maximize the power of new syntax.

A template lambda e.g. `x => x + 1` is equivalent with a normal template function `auto __lambda(T)(T x) { return x + 1; }`, but it still cannot have explicit template parameters. By using the optional template parameters on alias declaration, we can have complete alternative syntax against normal template function declarations.

```d
alias foo(alias pred) = x => pred(x) + 1;
// is equivalent with:
tempalte foo(alias pred) { auto foo(T)(T x) { return pred(x) + 1; } }
```